### PR TITLE
Add loading spinner to "Add to owncloud"

### LINKED ISSUE #23106: "Add to owncloud" should offer some kind of visual feedback
### Steps to reproduce
1. Follow a public link
2. Click on "Add to owncloud
3. Type 666 and "Enter"
4. Wait

### Expected behaviour

There should be some visual feedback that the query is being executed. It's usually a spinner. In this case, it could be in the middle of the field.

### Actual behaviour

Nothing happens for 20-30 seconds. The user gets confused and thinks ownCloud is broken.
After a while, a message is shown letting the user know that that URL doesn't lead to a valid share.

There is also the case against endless spinners as it's not a clear indication that things are working, but it's better than nothing I think.

@jancborchardt

### PR DESCRIPTION
**Pull request by ErikPel**
Fixes #23106

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @oparoz @jancborchardt

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by jancborchardt** [comment]
Up for a review @owncloud/designers @oparoz?

**Comment by MorrisJobke** [comment]
@jancborchardt Why didn't you reviewed it? ;)

**Comment by skjnldsv** [comment]
Ok here! :+1:

**Comment by jancborchardt** [comment]
@ErikPel this needs a rebase too, then we can merge :)

File: apps/files_sharing/js/public.js [hunk 1]
```diff
@@ -292,6 +292,21 @@ OCA.Sharing.PublicApp = {
 	},
 
 	_saveToOwnCloud: function (remote, token, owner, ownerDisplayName, name, isProtected) {
+		var toggleLoading = function() {
+			var iconClass = $('#save-button-confirm').attr('class');
+			var loading = iconClass.indexOf('icon-loading-small') !== -1;
+			if(loading) {
+				$('#save-button-confirm').removeClass("icon-loading-small");
```
Comment (DeepDiver1975) [inline apps/files_sharing/js/public.js]: Why not get rid of the duplicate  ?operation
```js

$('#save-button-confirm')
    .removeClass("icon-loading-small")
    .addClass("icon-confirm");

**Comment by jancborchardt** [comment]
@ErikPel wanna optimize the JS? Then we could merge. :)

**Comment by ErikPel** [comment]
@jancborchardt done

**Comment by jancborchardt** [comment]
@ErikPel thanks a lot! :)

**Pull request merged**